tests/unittests/tests-uri_parser/tests-uri_parser.c

```diff
@@ -21,599 +21,567 @@
+ * arn't identical.
+ */
+static int _compare_string_buffer(const char * name, const char * input_uri, const char * expected_str, const char * actual_str, size_t actual_len)
 {
-    uri_parser_result_t ures;
-    for (unsigned i = 0; i < ARRAY_SIZE(validate_uris); i++) {
-        int res = uri_parser_process_string(&ures, validate_uris[i].uri);
-        TEST_ASSERT_EQUAL_INT(validate_uris[i].full_uri,
-               uri_parser_is_absolute_string(validate_uris[i].uri));
-        TEST_ASSERT_EQUAL_INT(validate_uris[i].expected, res);
-        if (res == 0) {
-            VEC_CHECK_STR(scheme, i, _failure_msg);
-            VEC_CHECK_STR(userinfo, i, _failure_msg);
-            VEC_CHECK_STR(host, i, _failure_msg);
-            VEC_CHECK_STR(ipv6addr, i, _failure_msg);
-            VEC_CHECK_STR(zoneid, i, _failure_msg);
-            VEC_CHECK_STR(port_str, i, _failure_msg);
-            VEC_CHECK_INT(port, i, _failure_msg);
-            VEC_CHECK_STR(path, i, _failure_msg);
-            VEC_CHECK_STR(query, i, _failure_msg);
+    if((actual_len == strlen(expected_str)) && (strncmp(actual_str, expected_str, actual_len) == 0)) {
+        return 0;
+    }
+    printf("\nWith given input uri '%s', expected %s '%s' with length '%d' but got '%.*s' with length '%d'\n",
+            input_uri, name, expected_str, strlen(expected_str), actual_len, actual_str, actual_len);
+    return -1;
+}
+
+/* 
+ * Given a simple valid uri, this test checks that the parsing returns success
+ */
+static void _successfull_parsing_of_valid_uri(void) {
+    char *failure_msg = "Failure: The uri_parser failed to parse a correct and valid uri.";
+    int expected_ret = 0;  /* Indicates a successfull parsing */
+
+    char *simple_uri_00 = "coap://example.org/foo/bar";
+    char *simple_uri_01 = "ftp://riot-os.org/bar/foo";
+    char *simple_uri_02 = "ftp://riot-os.org:99/bar/foo";
+    char *simple_uri_03 = "http://riot-os.org:99/bar/foo";
+    char *simple_uri_04 = "https://riot-os.org/";
+
+    /* these are the old test vectors which contained valid uri.
+     * The invalid uri are in the next test. The numbering represents the 
+     * sequence / order in the old test.
+     */
+    char *simple_uri_05 = "coap://RIOT:test@[fe80:db8::1%tap0]:5683/.well-known/core?v=1";
+    char *simple_uri_07 = "coap://[fe80::1]/foo%20bar";
+    char *simple_uri_08 = "/.well-known/core?v=1";
+    char *simple_uri_09 = "coap://R@[2001:db8::1]:5/v=1";
+    char *simple_uri_10 = "coap://R@[2001:db8::1]:5/:v=1";
+    char *simple_uri_11 = "cap://R@[2001:db8::1]:5/?v=1";
+    char *simple_uri_12 = "oap://Y2001:db8::1]:5/av=1";
+    char *simple_uri_17 = "//Rb[ʰ00J:d/5v=0";
+    char *simple_uri_19 = "coap:///R@[2008::1]:5own//R@[2008::1]:5own/?v=1";
+    char *simple_uri_20 = "coaP://R/RZ[2001[8:01[8::1]:5o:1]:5oTMv=1";
+    char *simple_uri_21 = "coap://R@////////////////7///v=1";
+    char *simple_uri_22 = "coa[:////[2001:db5ow:5own/Ov=1";
```

Comment:
A ton of the old test vectors are just like "mh okay? What exactly is being tested?". If you got any ideas, please drop them so I can rename them from `simple_uri` to something more helpful.

Reply:
I guess @cgundogan is the best person to ask here.